lib/adapter.js

```diff
@@ -155,6 +159,7 @@ class MochaAdapter {
         if (err) {
             this.runner.lastError = err
         }
+        process.emit(event, message)
```

Comment:
Yeah, this is something I'm unsure about. It's no wonder you didn't understand why - I hadn't pushed the equivalent change to webdriverio yet! See: https://github.com/webdriverio/webdriverio/commit/7057bdf4ee51fee45a22d6a9055ef364006a6511

The framework adapter events are associated with a particular test, but events from the webdriverio library (such as `command`, `result`, etc.) are not. This change means that the test runner always keeps track of the current test (https://github.com/webdriverio/webdriverio/pull/896/files#diff-b3b53682a18f203ac8d29b0e277cad26R105), and that any non-mocha events can be associated with that test. It means that the Junit stdout block for a particular test can contain just the commands/results which occurred during that test.

If you can think of a better way of doing this, I'd love to know.